backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandBL.java

```diff
@@ -1456,7 +1456,7 @@ public void handleReversalForInvoice(final org.compiere.model.I_C_Invoice invoic
 					note = "@C_InvoiceLine@  @QtyInvoiced@ = " + il.getQtyInvoiced() + " @IsCreditedInvoiceReinvoicable@='Y'; ignoring overlap, because credit memo";
 				}
 
-				else if (creditMemo && !creditedInvoiceReinvoicable && creditedInvoiceIsReversed)
+				else if (creditMemo && !creditedInvoiceReinvoicable)
```

Comment:
` && creditedInvoiceIsReversed` was there for a reason..
From the variable naming I assume that it was there to cover the case that you have an invoice, then create a credit-memo and then revert the invoice.

Update: I used git blame and found that this is the reason: https://github.com/metasfresh/metasfresh/issues/6540
Luckily Ruxi made a test in InvoiceCandBLHandlCompleteAndReversalTest that now fails :-P
Please check out those tests and add one for this issue's case as well, when it's fixed

Reply:
Fixed.